mne/io/utils.py

```diff
@@ -0,0 +1,24 @@
+import numpy as np
+from .base import _mult_cal_one
+
+
+def _read_segments_file(raw, data, idx, fi, start, stop, cals, mult,
+                        dtype='<i2', n_bytes=2):
+    """Read a chunk of raw data"""
+    nchan = raw.info['nchan']
+    data_offset = raw.info['nchan'] * start * n_bytes
+    data_left = (stop - start) * nchan
+    # Read up to 100 MB of data at a time.
+    blk_size = min(data_left, ((100000000 // n_bytes) // nchan) * nchan)
+
+    with open(raw._filenames[fi], 'rb', buffering=0) as fid:
+        fid.seek(data_offset)
+        # extract data in chunks
+        for blk_start in np.arange(0, data_left, blk_size) // nchan:
+            blk_size = min(blk_size, data_left - blk_start * nchan)
```

Comment:
It is a bit strange to have `data_left`, `blk_size` in `bytes` samples*chs units while `blk_start` is in `samples` units. Can you unify it all?

It's also not good to reuse the variable name `blk_size` here. Even though I think you are guaranteed it should only be changed on the last iteration of the loop, it takes some thinking to determine this fact (and a future person looking at the code might not realize it).